internal/generator/backend/template/zkpschemes/plonk/plonk.marshal.go.tmpl

```diff
@@ -98,16 +99,16 @@ func (pk *ProvingKey) WriteTo(w io.Writer) (n int64, err error) {
 	// encode the size (nor does it convert from Montgomery to Regular form)
 	// so we explicitly transmit []fr.Element
 	toEncode := []interface{}{
-		([]fr.Element)(pk.Ql),
-		([]fr.Element)(pk.Qr),
-		([]fr.Element)(pk.Qm),
-		([]fr.Element)(pk.Qo),
-		([]fr.Element)(pk.CQk),
-		([]fr.Element)(pk.LQk),
-		([]fr.Element)(pk.S1Canonical),
-		([]fr.Element)(pk.S2Canonical),
-		([]fr.Element)(pk.S3Canonical),
-		pk.Permutation,
+		([]fr.Element)(pk.trace.Ql.Coefficients()),
```

Comment:
shouldn't `plonk.Trace` object implement directly the `WriteTo` and `ReadFrom` methods?

Reply:
+ instead of serializing the coeff only, shouldn't we serialize also the flags that indicates the form ? (instead of relying on the deserialization code to manually set something that matches the naming convention)

Reply:
Yes we should have methods for serialising iop.Polynomial. In fact in gnark-crypto an iop.Polynomial should inherit from Vector... For the moment for gnark I think it's enough to do it this way, the proving is (and should not be mutated) anyway